src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,128 @@
+package com.leeyiyuan.storage;
+
+
+import com.leeyiyuan.storage.format.DeadlineTaskFormatter;
+import com.leeyiyuan.storage.format.EventTaskFormatter;
+import com.leeyiyuan.storage.format.TaskFormatException;
+import com.leeyiyuan.storage.format.TaskFormatter;
+import com.leeyiyuan.storage.format.TaskParseException;
+import com.leeyiyuan.storage.format.TodoTaskFormatter;
+import com.leeyiyuan.task.Task;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents a storage for tasks persistence.
+ */
+public class Storage {
+
+    /** Path of file used to back Storage. */
+    protected String filePath;
+
+    /** Fixed list of TaskFormatters. */
+    protected ArrayList<TaskFormatter> taskFormatters;
+
+    /**
+     * Constructs a Storage using a file path.
+     *
+     * @param filePath Path of file used to back Storage.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.taskFormatters = new ArrayList<TaskFormatter>();
```

Comment:
(written for tutorial activity)

there may be a better abstraction for the TaskFormatters (instead of having to add a new TaskFormatter every time there is a new type of Task)

Reply:
Hmm... could you come up with an example of how to better abstract this? Domain knowledge has to be specified somewhere, and I thought that with this abstraction it becomes convenient to specify task specific domain knowledge in an isolated, loosely coupled way through extending `TaskFormatter`. Let me know what you think!